pkg/operator/deploy/deploy.go

```diff
@@ -128,6 +129,16 @@ func (o *operator) resources() ([]runtime.Object, error) {
 		return nil, err
 	}
 
+	var monitoringEndpoint string
+	switch o.env.Environment().Name {
+	case "AzurePublicCloud":
```

Comment:
I am vaguely tempted by
```golang
	switch o.env.Environment() {
	case &azure.PublicCloud:
```

-- it avoids the non-const string.

Reply:
That wouldn't work. `azure.EnvironmentFromName` (used to set `o.env.environment`) returns `Environment`, not `*Environment`. The memory address returned by `o.env.Environment()` is therefore different from `&azure.PublicCloud`.

I can get the string value from `azure.PublicCloud`, though. That would be cleaner.